### PR TITLE
Feature/209 thumb preview in entity list

### PR DESCRIPTION
**Pull request by pieroit**

### CODE REVIEW

File: src/admin/class-wordlift-admin-entity-list.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Customization of the entity list in wp-admin/edit.php
+ * 
+ * @since 3.3.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
```
Comment (ziodave) [inline src/admin/class-wordlift-admin-entity-list.php]: in our case this is superfluous being just a class definition

File: src/admin/class-wordlift-admin-entity-list.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		
+	}
+
+	/**
+	 * Register custom columns for entity listing in backend
+	 * @see https://codex.wordpress.org/Plugin_API/Action_Reference/manage_posts_custom_column
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param array $columns the current column.
```
Comment (ziodave) [inline src/admin/class-wordlift-admin-entity-list.php]: is this one column or all the columns?

File: src/admin/class-wordlift-admin-entity-list.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	 *
+	 * @return array Enhanced columns array.
+	 */
+	public function register_custom_columns( $columns ) {
+		
+		// take away first column
```
Comment (ziodave) [inline src/admin/class-wordlift-admin-entity-list.php]: can you explain in the comments why we take out the first column and then add it back?

File: src/admin/class-wordlift-admin-entity-list.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		);
+
+		return $columns;		
+	}
+
+	/**
+	 * Render custom columns
+	 * @see https://codex.wordpress.org/Plugin_API/Action_Reference/manage_$post_type_posts_custom_column
+	 *
+	 * @since 3.2.0
```
Comment (ziodave) [inline src/admin/class-wordlift-admin-entity-list.php]: should be 3.3.0 right?

File: src/admin/class-wordlift-admin-entity-list.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+				break;
+			
+			case 'wl_column_thumbnail':
+				
+				$img_size  = 50;
+				$edit_link = get_edit_post_link( $entity_id );
+				$thumb     = get_the_post_thumbnail( $entity_id, array( $img_size, $img_size ) );
+				
+				if( ! $thumb ) {
+					$thumb = "<img src='" . WL_DEFAULT_THUMBNAIL_PATH . "' width='$img_size' />";
```
Comment (ziodave) [inline src/admin/class-wordlift-admin-entity-list.php]: better avoid reusing variables

File: src/admin/class-wordlift-admin-entity-list.php (hunk 2 above)
Comment (mcolacino) [inline src/admin/class-wordlift-admin-entity-list.php]: It's a separated service for all changes related to entity listing enhancement (so all the columns)

File: src/admin/class-wordlift-admin-entity-list.php (hunk 4 above)
Comment (mcolacino) [inline src/admin/class-wordlift-admin-entity-list.php]: the method was migrated in the new class ... but yes it should be 3.3.0

**Comment by mcolacino** [comment]
I'd manage this as a class constant

File: src/admin/class-wordlift-admin-entity-list.php (hunk 2 above)
Comment (mcolacino) [inline src/admin/class-wordlift-admin-entity-list.php]: Do we need a blank *__construct()* method?

Comment (ziodave) [inline src/admin/class-wordlift-admin-entity-list.php]: shall we update the comment `the current column` then?

File: src/admin/class-wordlift-admin-entity-list.php (hunk 5 above)
Comment (mcolacino) [inline src/admin/class-wordlift-admin-entity-list.php]: Here I'd set *$img_size* as a class constant

**Pull request merged**